sdk/src/main/java/com/hedera/hashgraph/sdk/Transaction.java

```diff
@@ -49,6 +48,9 @@
     // ]
     int nextTransactionIndex = 0;
 
+    private List<PublicKey> publicKeys = new ArrayList<>();
```

Comment:
Can we make this a Set instead? Would remove the looped equality check in "already signed".

Reply:
We need this to be ordered the same as `signers` since so we know the signer for `publicKeys.get(i)` is `signers.get(i)`. We _can_ add another field which stores the public keys in a set, and only use that set for the already signed check.